src/mon/OSDMonitor.cc

```diff
@@ -7732,7 +7732,7 @@ bool OSDMonitor::preprocess_pool_op(MonOpRequestRef op)
 
   if (!osdmap.get_pg_pool(m->pool)) {
     dout(10) << "attempt to operate on non-existent pool id " << m->pool << dendl;
-    _pool_op_reply(op, 0, osdmap.get_epoch());
+    _pool_op_reply(op, -ENOENT, osdmap.get_epoch());
```

Comment:
@tchaikov This change cause a lot of failures, such as:
http://qa-proxy.ceph.com/teuthology/yuriw-2016-08-16_17:06:30-rados-wip-yuri-testing_2016_08_15-distro-basic-smithi/368601/teuthology.log
http://qa-proxy.ceph.com/teuthology/yuriw-2016-08-16_17:06:30-rados-wip-yuri-testing_2016_08_15-distro-basic-smithi/368655/teuthology.log